org/mozilla/jss/netscape/security/x509/CertificateExtensions.java

```diff
@@ -253,19 +255,19 @@ public void delete(String name) throws IOException {
     }
 
     public Enumeration<String> getNames() {
-        return map.keys();
+        return Collections.enumeration(map.keySet());
     }
 
     /**
      * Return an enumeration of names of attributes existing within this
      * attribute.
      */
     public Enumeration<Extension> getAttributes() {
-        return (map.elements());
+        return new Vector(map.values()).elements();
     }
 
     public Enumeration<String> getAttributeNames() {
-        return (map.keys());
+        return new Vector(map.keySet()).elements();
```

Comment:
Any reason why we are returning a `Vector`? Can we reuse the getName() method? Like this?
````
return getNames();
````